File: kod/SocialMediumForMusicians/Data/Models/Message.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SocialMediumForMusicians.Data.Models
{
    [Table("Messages")]
    public class Message
    {
        [Key]
        public int Key { get; set; }

        [Required]
        public string Id { get; set; }

        [Required]
        public string AuthorId { get; set; }

        [Required]
        public string RecipentId { get; set; }

        public User Author { get; set; }

        public User Recipent { get; set; }

        [Required]
        public string Content { get; set; }

        public bool Read { get; set; }

        [Required]
        public DateTime SentAt { get; set; }
    }
}

File: kod/SocialMediumForMusicians/ClientApp/src/models/message.ts
export interface Message {
    id?: string;
    authorId: string;
    recipentId: string;
    content: string;
    read: boolean;
    sentAt: Date;
    authorName?: string;
    authorImgFilename?: string;
}

export interface EmailMessage {
    id?: string;
    authorEmail: string;
    recipentId: string;
    content: string;
    read: boolean;
    sentAt: Date;
}
